File: yukicoder/487_2017Calculation(2017の計算)/Main.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        int M = sc.nextInt();

        long result = 1;

        for (int i = 0; i < 2017; i++) {
            result = result * 2017 * 2017 % M;
        }

        result += 2017;
        result %= M;

        System.out.println(result);

    }

}

File: yukicoder/487_2017Calculation(2017の計算)/Main.cpp
#include <iostream>

using namespace std;

int main() {

    int M;
    cin >> M;

    long result = 1;

    for (int i = 0; i < 2017; i++) {
        result = result * 2017 * 2017 % M;
    }

    result += 2017;
    result %= M;

    cout << result << endl;

    return 0;

}
